File: app/src/main/java/gov/census/cspro/csentrylauncher/MainActivity.java
package gov.census.cspro.csentrylauncher;

import android.content.ActivityNotFoundException;
import android.content.ComponentName;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;

public class MainActivity extends AppCompatActivity {

    private EditText mEditTextPffFile;
    private EditText mEditTextCaseId;
    private Spinner mSpinnerMode;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mEditTextPffFile = findViewById(R.id.editTextPff);
        mEditTextCaseId = findViewById(R.id.editTextCaseId);
        mSpinnerMode = findViewById(R.id.spinnerMode);

        // Only enable for modify mode
        mEditTextCaseId.setEnabled(mSpinnerMode.getSelectedItemPosition() == 1);

        mSpinnerMode.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int pos, long id) {
                // Only enable for modify mode
                mEditTextCaseId.setEnabled(pos == 1);
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {
            }
        });

        Button buttonLaunch = findViewById(R.id.buttonLaunch);
        buttonLaunch.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                final String pffFile = mEditTextPffFile.getText().toString();
                final boolean addMode = mSpinnerMode.getSelectedItemPosition() == 0;
                final String startCaseId = addMode ? null : mEditTextCaseId.getText().toString();
                launchCSEntry(pffFile, addMode, startCaseId);
            }
        });
    }

    private void launchCSEntry(String pffFile, boolean addMode, String startCaseId) {

        try {
            Intent intent = new Intent();
            intent.setComponent(new ComponentName("gov.census.cspro.csentry", "gov.census.cspro.csentry.ui.EntryActivity"));
            if (addMode) {
                intent.putExtra("StartMode", "add");
            } else {
                intent.putExtra("Key", startCaseId);
            }
            intent.putExtra("PffFilename", pffFile);
            intent.putExtra("OperatorID", "josh");
            startActivity(intent);
        } catch (ActivityNotFoundException e)
        {
            AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(this);
            alertDialogBuilder.setMessage(R.string.activity_not_found).show();
        }
    }
}

File: app/src/main/java/gov/census/cspro/csentrylauncher/MainActivity.kt
package gov.census.cspro.csentrylauncher

import android.content.ActivityNotFoundException
import android.content.ComponentName
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import androidx.activity.viewModels
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import java.util.*

class MainActivity : AppCompatActivity() {
    private lateinit var mEditTextPffFile: EditText
    private lateinit var mEditTextStaffName: EditText
    private val casesViewModel: CSProCasesViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mEditTextPffFile = findViewById(R.id.editTextPff)
        mEditTextStaffName = findViewById(R.id.editTextStaffName)

        findViewById<Button>(R.id.buttonLaunchViaIntent).setOnClickListener {
            launchCSEntryViaIntent(mEditTextPffFile.text.toString(), mEditTextStaffName.text.toString())
        }

        findViewById<Button>(R.id.buttonLaunchViaDeepLink).setOnClickListener {
            launchCSEntryViaDeepLink(mEditTextPffFile.text.toString(), mEditTextStaffName.text.toString())
        }

        // RecyclerView that displays list of cases from CSPro
        // Cases are added when CSPro calls the GetCSProResultActivity
        val recyclerView = findViewById<RecyclerView>(R.id.recyclerview)
        val adapter = CSProCaseListAdapter(this)
        recyclerView.adapter = adapter
        recyclerView.layoutManager = LinearLayoutManager(this)
        casesViewModel.allCases.observe(this) { cases ->
            // Update the cached copy of the words in the adapter.
            cases?.let { adapter.setCases(it) }
        }
    }

    private fun launchCSEntryViaIntent(pffFile: String, staffName: String) {
        try {
            val intent = Intent()
            intent.component = ComponentName("gov.census.cspro.csentry", "gov.census.cspro.csentry.ui.EntryActivity")
            intent.putExtra("PffFilename", pffFile)
            intent.putExtra("STAFF_NAME", staffName)
            intent.putExtra("Key", UUID.randomUUID())
            startActivity(intent)
        } catch (e: ActivityNotFoundException) {
            val alertDialogBuilder = AlertDialog.Builder(this)
            alertDialogBuilder.setMessage(R.string.activity_not_found).show()
        }
    }

    private fun launchCSEntryViaDeepLink(pffFile: String, staffName: String) {
        val uri = Uri.parse("https://csprousers.org/pff")
            .buildUpon()
            .appendPath(pffFile)
            .appendQueryParameter("STAFF_NAME", staffName)
            .appendQueryParameter("Key", UUID.randomUUID().toString())
            .build()

        val intent = Intent(Intent.ACTION_VIEW, uri)
        startActivity(intent)
    }
}
